### PR TITLE
Added cache invalidation when the playground have a new build.

### PR DESCRIPTION
**Pull request by Kuinox**
Added a js dep at build time, this is an esbuild plugin that inject the date at build time in a class.
We could easily write this ourselves, but I have 0 motivation to understand more yet another JS pipeline.

### CODE REVIEW

File: src/Draco.Editor.Web/app/ts/cache.ts [hunk 1]
```diff
@@ -15,8 +15,28 @@ export function getDownloadViewElement() {
...
+        if (result == null || cachedDate != buildDate) {
+            console.log('Cache nuked.');
+            await caches.delete('assembly-cache');
+        }
+    } catch (e) {
+        console.log('Could not open cache: ', e);
+    }
+    try {
+        const cache = await caches.open('assembly-cache');
+        await cache.put('appBuildDate', new Response(buildDate));
```
Comment (Binto86) [inline src/Draco.Editor.Web/app/ts/cache.ts:33]: Can't this be in the previous try catch?

Comment (Binto86) [reply src/Draco.Editor.Web/app/ts/cache.ts:33]: Do i understand correctly that this fails if you nuked the cache before? if yes, where do you put the new date after you loaded the new compiler version into the cache?

Comment (Kuinox) [reply src/Draco.Editor.Web/app/ts/cache.ts:33]: In certain scenario, calling caches.open just doesn't work some browser, for example when you are in private mode.

File: src/Draco.Editor.Web/app/ts/cache.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { blobToBase64 } from './helpers.js';
-
+import { buildDate } from './metadata.js';
 const elements = [];
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/app/ts/cache.ts:3]:
```suggestion
import { buildDate } from './metadata.js';

const elements = [];
```

File: src/Draco.Editor.Web/app/ts/cache.ts [hunk 3]
```diff
@@ -40,9 +55,9 @@ async function downloadAssembly(dlPath: string, asset: unknown): Promise<void> {
     } catch (e) {
         console.log('Could not open cache: ', e);
     }
-    console.log('Cache miss:'+dlPath);
+    console.log('Cache miss:' + dlPath);
```
Comment (LPeter1997) [inline src/Draco.Editor.Web/app/ts/cache.ts:58]: We could use interpolation and add a space in between

**Review by LPeter1997 — APPROVED** [review]

**Pull request merged**